src/server/game/Spells/Spell.cpp

```diff
@@ -7961,6 +7961,17 @@ bool WorldObjectSpellTargetCheck::operator()(WorldObject* target)
             default:
                 break;
         }
+
+        switch (_objectType)
+        {
+            case TARGET_OBJECT_TYPE_CORPSE:
+            case TARGET_OBJECT_TYPE_CORPSE_ALLY:
+            case TARGET_OBJECT_TYPE_CORPSE_ENEMY:
+                if (unitTarget->IsAlive())
```

Comment:
possible crash: you gotta check if `unitTarget` is != null, its a cast `Unit* unitTarget = target->ToUnit();` which may result in null value.

You might just put it inside the `if (unitTarget)` block above. Even above the other switch-case block for even faster reaction.

Reply:
It's hard to tell due to github collapsing code areas, but this block is actually inside a `if (unitTarget)` block, so there's no need to check that.